operate/wallet/wallet_recovery_manager.py

```diff
@@ -81,35 +140,190 @@ def initiate_recovery(self, new_password: str) -> t.Dict:
+                owners = get_owners(ledger_api=ledger_api, safe=safe)
+                backup_owners = set(owners) - {wallet.address} - {new_wallet.address}
+
+                if len(backup_owners) != 1:
+                    self.logger.warning(
+                        f"[WALLET RECOVERY MANAGER] Safe {safe} on {chain.value} has unexpected number of backup owners: {len(backup_owners)}."
+                    )
+
+                for backup_owner in backup_owners:
+                    balances[chain_str].setdefault(backup_owner, {})
+                    balances[chain_str][backup_owner][ZERO_ADDRESS] = get_asset_balance(
+                        ledger_api=ledger_api,
+                        asset_address=ZERO_ADDRESS,
+                        address=backup_owner,
+                        raise_on_invalid_address=False,
+                    )
+                    requirements[chain_str].setdefault(backup_owner, {}).setdefault(
+                        ZERO_ADDRESS, 0
+                    )
+                    if new_wallet.address not in owners:
+                        requirements[chain_str][backup_owner][
+                            ZERO_ADDRESS
+                        ] += DEFAULT_RECOVERY_TOPUPS[chain][ZERO_ADDRESS]
+                        pending_backup_owner_swaps.setdefault(chain_str, [])
+                        if safe not in pending_backup_owner_swaps[chain_str]:
+                            pending_backup_owner_swaps[chain_str].append(safe)
+
+        refill_requirements = ChainAmounts.shortfalls(
+            requirements=requirements, balances=balances
+        )
+        is_refill_required = any(
+            amount > 0
+            for address in refill_requirements.values()
+            for assets in address.values()
+            for amount in assets.values()
+        )
 
         return {
-            "id": bundle_id,
-            "wallets": output,
+            "balances": balances,
+            "total_requirements": requirements,
+            "refill_requirements": refill_requirements,
+            "is_refill_required": is_refill_required,
+            "pending_backup_owner_swaps": pending_backup_owner_swaps,
+        }
+
+    def status(self) -> t.Dict[str, t.Any]:
+        """Get recovery status."""
+        bundle_id = self.data.last_prepared_bundle_id
+        if not bundle_id:
+            return {
+                "prepared": False,
+                "bundle_id": bundle_id,
+                "has_swaps": False,
+                "has_pending_swaps": False,
+            }
+
+        (
+            _,
+            num_safes_with_new_wallet,
+            num_safes_with_old_wallet,
+            num_safes_with_both_wallets,
+        ) = self._get_swap_status(bundle_id)
+
+        return {
+            "prepared": bundle_id is not None,
+            "bundle_id": bundle_id,
+            "has_swaps": num_safes_with_new_wallet + num_safes_with_both_wallets > 0,
+            "has_pending_swaps": num_safes_with_old_wallet > 0,
```

Comment:
pending swaps means the safes where the new wallet is not the owner yet, right?
But here, we check the number of current safes with the old wallet. An owner can still keep the ownership of the old wallet, while also staying the owner of the new wallet, or not?

Reply:
so the _get_swap_stats returns 4 values:
_,
num_safes_with_new_wallet,  -> ONLY with new wallet and not old wallet
num_safes_with_old_wallet, -> ONLY with old wallet and not new wallet
num_safes_with_both_wallets, -> with BOTH wallets (new and old)

I know it's a bit confusing, if you have any suggestion to improve...

Reply:
yeah I understood that, I meant that `has_pending_swaps` seems like should be formulated as `num_safes_with_new_wallet < num_safes`. Wdyt?

Reply:
hmm don't quite agree. `old` or `new` are the master EOAs. Safes are not old.